lib/chai/core/assertions.js

```diff
@@ -528,10 +528,25 @@ module.exports = function (chai, _) {
    */
 
   Assertion.addProperty('empty', function () {
-    var obj = flag(this, 'object');
-    new Assertion(obj).to.exist;
+    var val = flag(this, 'object')
+      , itemsCount;
+
+    switch (_.type(val)) {
+      case 'array':
+      case 'string':
+        itemsCount = val.length;
+        break;
+      case 'function':
+        throw new TypeError('.empty() was passed a function');
+      default:
+        if (val !== Object(val)) {
+          throw new TypeError('.empty() was passed non-string primitive');
+        }
+        itemsCount = Object.keys(val).length;
+    }
+
     this.assert(
-        Object.keys(Object(obj)).length === 0
+        0 === itemsCount
```

Comment:
Is there ESLint config we can add to the project? This will greatly improve contribution/review experience: yoda comparisons and comma-first are not quite popular and it would be awesome to automate codestyle checks.